docs/pages/auth-concept/authentication.md

```diff
@@ -0,0 +1,181 @@
+| ---------- | --------------------------------------------------------------- |
+| done       | The done function that tells the server to proceed.             |
+| authProps  | The authentication parameters recieved from the client.         |
+| serverName | The name of the server/broker from which the event was emitted. |
+| doc        | The parsedAsyncAPI schema                                       |
+
+### done()
+
+The `done` parameter in the `serverAuth` function allows the broker/server to know what to do next depending on the boolean value you pass to it.
+
+```js
+/* websocket.js */
+
+export async function serverAuth({ authProps, done }) {
+  // done(true)
+  //done(false, 401, "Unauthorized")
+  // done(false)
+}
+```
+
+When `true` is passed to the done parameter, the server/broker knows to go ahead and allow the client to connect, which means authentication has succeeded. However if the `done` parameter is called with `false` then the server knows to throw an error message and reject the client, which means authenticatio has failed.
+
+`done()` should always be the last thing called in a `serverAuth` function, Glee won't execute any logic beyond the `done()` call.
+
+### authProps
+
+`authProps` implements a couple of methods that allows the server to retrieve the authentication parameters from the client, below are the current available methods;
+
+```js
+export async function serverAuth({ authProps, done }) {
+  //some network request
+
+  authProps.getOauthToken()
+  authProps.getHttpAPIKeys('api_key')
+  authProps.getToken()
+  authProps.getUserPass()
+
+  // done(false, 401, "Unauthorized");
+  done(false)
+}
+```
+
+| Method                 | Description                                                                                      |
+| ---------------------- | ------------------------------------------------------------------------------------------------ |
+| `getOauthToken()`      | returns the oauth authentication parameter                                                       |
+| `getHttpAPIKeys(name)` | returns the HttpAPIKeys parameter with the specified name from either headers or query parameter |
+| `getToken()`           | returns the http bearer token parameter                                                          |
+| `getUserPass()`        | returns username and password parameters                                                         |
+
+## Client Authentication in Glee
+
+The `clientAuth` function also takes an argument, and it's argument can be destructured as follows
+
+| Attribute      | Description                                                                           |
+| -------------- | ------------------------------------------------------------------------------------- |
+| parsedAsyncAPI | The parsedAsyncAPI schema.                                                            |
+| serverName     | The name of the server/broker from with the authentication parameters are being sent. |
```

Comment:
@Souvikns @oviecodes does the auth function really need arguments? what would be the use case of parsedAsyncAPI and the user already has the serverName since they created a file with the name of the server right?